### DISCUSSION TITLE
使用QCefview打开登录网址，登陆成功后本来应该覆盖显示的页面，却弹出的新窗口

### DISCUSSION THREAD

**Discussion by xiaomingtongxue4221** [CefView/QCefView#298] [Q&A]
打开一个登录网页链接，输入账号密码正常登录后，在cef（使用libcef_dll_wrapper库）中测试是新的页面覆盖旧的页面，且正常显示。同样操作放到QCefview中调用，会打开新的窗口，此时新窗口有效，而旧窗口已经失效，如果操作旧窗口的对象指针也会报错，直接崩溃。此种情况是不是需要设置什么状态或者属性，为什么cef中正常，到QCefview中就出问题了，有大佬能帮忙看下吗？谢谢了。。。

**Comment by tishion** [comment]
最好给一个能复现的站点地址

**Reply by xiaomingtongxue4221** [reply]
地址目前还没有。。我还是给您描述一下吧，就是在使用cef库的项目中打开这个网页登录成功后，会直接在当前窗口跳转到新的页面，没有触发onBeforePopup回调。而在使用QCefview的项目中，同样的网页登录成功后，触发了onBeforePopup回调，且类型是QCefView::CefWindowOpenDisposition::CefWindowOpenDispositionNewPopup，然后弹出新的窗口，此时如果点击最开始的窗口上的任何控件都是无效的（感觉应该是最开始的窗口被析构了），此时用最开始窗口指针调用navigateToUrl就直接崩溃了

**Reply by tishion** [reply]
https://github.com/CefView/QCefView/pull/303 
可以试试这个修复，macOS和Linux没有测试过
